src/main/java/map/FileData.java

```diff
@@ -0,0 +1,26 @@
+package map;
+
+public class FileData {
```

Comment:
About records - it's possible :). It's the way with immutable model classes 

In our existing prod java code using different approaches, the similar as it, so going in this way